src/Jackett.Common/Utils/StringUtil.cs

```diff
@@ -226,16 +226,16 @@ public static IEnumerable<int> AllIndexesOf(this string source, string value)
         /// </summary>
         public static IEnumerable<string> FindSubstringsBetween(this string source, char opening, char closing, bool includeOpeningAndClosing)
         {
-            var openingIndexes = source.AllIndexesOf(opening).ToList();
+            var openingIndexes = source.AllIndexesOf(opening).OrderByDescending(_ => _).ToList();
             var closingIndexes = source.AllIndexesOf(closing);
 
             foreach (var closingIndex in closingIndexes.OrderBy(_ => _))
             {
-                var potentialOpeningIndexes = openingIndexes.Where(x => x < closingIndex);
-                if (!potentialOpeningIndexes.Any())
+                var potentialOpeningIndex = openingIndexes.Where(x => x < closingIndex).Cast<int?>().FirstOrDefault();
```

Comment:
`Any` + `First` relaced with `FirstOrDefault` to avoid double filtering.
As this is an enumeration of int and it's default value (0) can be a valid index, I added `Cast<int?>()`